Fix type annotation errors in torch.distributed.* directory

xuzhao9 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47534 Fix typing errors in torch.distributed.*, close issue #42967.
* #47533 Fix typing errors in torch.distributed.nn.* directory.
* #47532 Fix typing errors in torch.distributed.distributed_c10d.*
* **#47531 Fix type annotation errors in torch.distributed.* directory**

Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24952499](https://our.internmc.facebook.com/intern/diff/D24952499)

xuzhao9 committed:
Fix type annotation errors in torch.distributed.* directory

Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

xuzhao9 commented on torch/csrc/distributed/rpc/testing/init.cpp:
```diff
@@ -97,6 +98,11 @@ PyObject* faulty_agent_init(PyObject* _unused, PyObject* noargs) {
           (const WorkerInfo& (ProcessGroupAgent::*)(const std::string&)const) &
               ProcessGroupAgent::getWorkerInfo,
           py::call_guard<py::gil_scoped_release>())
+      .def(
```
Have to import this function, otherwise mypy will report in class FaultyProcessGroupRpcBackendOptions, "get_worker_info" has different signature than its super class.

walterddr commented on torch/_C/_distributed_rpc_testing.pyi:
```diff
@@ -0,0 +1,42 @@
+    @overload
+    def get_worker_info(self, workerName: str) -> WorkerInfo: ...
+    @overload
+    def get_worker_info(self, id: int) -> WorkerInfo: ...
```
i dont see this signature though

xuzhao9 replied:
> i dont see this signature though

This signature is added on line 101 in rpc/testing/init.cpp (see my comments in the diff). Mypy will complain if the subclass's overload functions have different signature than its parent class.

xuzhao9 committed:
Update on "Fix type annotation errors in torch.distributed.* directory"


Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix type annotation errors in torch.distributed.* directory"


Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix type annotation errors in torch.distributed.* directory"


Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24952499](https://our.internmc.facebook.com/intern/diff/D24952499)

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix type annotation errors in torch.distributed.* directory"


Summary:
This is part of a stack of PRs that fixes mypy typing errors in the torch.distributed.* directory.

Test Plan:

python test_type_hints.py -v TestTypeHints.test_run_mypy

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24952499](https://our.internmc.facebook.com/intern/diff/D24952499)

[ghstack-poisoned]
